projects/transformers/run_utils.py

```diff
@@ -721,6 +743,109 @@ def check_for_callback(model_args, class_of_callback):
     return has_callback, callback_instance
 
 
+def check_sparsity_callback(model, model_args):
+
+    is_sparse = False
+    for module in model.modules():
+        if isinstance(module, SparseWeightsBase):
+            is_sparse = True
+            break
+
+    if is_sparse:
+        has_rezero = check_for_callback(model_args, RezeroWeightsCallback)
+        assert has_rezero, "Finetuning sparse models without rezeroing weights"
+        " is prohibited"
+
+
+def check_eval_and_max_steps(training_args, train_dataset):
+    """
+    If you're supposed to load best model at end, but evaluate() never gets
+    called because eval_steps > number of training steps taken, you'll get
+    an error at the end of training. If this is the case, set eval_steps
+    equal to max_steps so it gets called at least once.
+    """
+    if training_args.load_best_model_at_end:
+        if training_args.max_steps == -1:
+            num_examples = training_args.num_train_epochs * len(train_dataset)
+            max_steps = num_examples // training_args.per_device_train_batch_size
+        else:
+            max_steps = training_args.max_steps
+        if max_steps < training_args.eval_steps:
+            logging.warning(
+                f"max_steps({max_steps}) < "
+                f"eval_steps({training_args.eval_steps}) "
+                "To avoid issues, setting eval steps equal to max_steps"
+            )
+            training_args.eval_steps = max_steps
+
+    return training_args
+
+
+def check_hp_compute_objective(model_args, task_name):
+
+    if hasattr(model_args, "hp_compute_objective"):
```

Comment:
In general it helps to reduce the depth of indents. You can take away at least one here by condensing the outermost if statement as
```
hp_compute_objective = getattr(model_args, "hp_compute_objective", None)
if hp_compute_objective is not None:
    ....
```

Reply:
Good point. Adding a default to getattr is something I'll use more often going forward.